### PR TITLE
Enable nullability in LinkLabel

### PR DESCRIPTION
**Pull request by gpetrou**
## Proposed changes

- Enable nullability in LinkLabel.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/7877)

### CODE REVIEW

**Review by dreddy-work — APPROVED** [review]
LGTM.

**Comment by RussKie** [comment]
@msftbot forget it

**Comment by RussKie** [comment]
There are few new `!` operators and I'd like to review each one of those.

**Review by RussKie — CHANGES_REQUESTED** [review]
Blocking so I can review `!` operators

**Comment by gpetrou** [comment]
> Blocking so I can review `!` operators

@RussKie any more thoughts on this PR?

File: src/System.Windows.Forms/src/System/Windows/Forms/LinkUtilities.cs [hunk 1]
```diff
@@ -158,7 +158,7 @@ public static LinkBehavior GetIELinkBehavior()
             return LinkBehavior.AlwaysUnderline;
         }
 
-        public static void EnsureLinkFonts(Font baseFont, LinkBehavior link, ref Font linkFont, ref Font hoverLinkFont, bool isActive = false)
+        public static void EnsureLinkFonts(Font baseFont, LinkBehavior link, ref Font? linkFont, ref Font? hoverLinkFont, bool isActive = false)
```
[the remark this answers is not in the document]
Comment (gpetrou) [reply src/System.Windows.Forms/src/System/Windows/Forms/LinkUtilities.cs:161]: Hmm, we do have null checks below, so I assume that these should be nullable. Note that this method is used in two other classes that are not annotated yet where `linkFont` and `hoverLinkFont` seem to be nullable as well.

[the remark this answers is not in the document]
Comment (RussKie) [reply src/System.Windows.Forms/src/System/Windows/Forms/LinkUtilities.cs:161]: https://github.com/gpetrou/winforms/blob/40be9862dcd56473e3ea255d741e1ab98181dbcc/src/System.Windows.Forms/src/System/Windows/Forms/LinkUtilities.cs#L161-L247

I don't see any codepaths that would result in either `linkFont` or `hoverLinkFont` being null at the end of the call.

It looks like I forgot `[AllowNull]` annotations, updated.

[the remark this answers is not in the document]
Comment (RussKie) [reply src/System.Windows.Forms/src/System/Windows/Forms/LinkUtilities.cs:161]: image

**Review by RussKie — APPROVED** [review]

**Comment by RussKie** [comment]
Thank you

**Pull request merged**